unison-cli/src/Unison/CommandLine.hs

```diff
@@ -148,12 +150,12 @@ parseInput getRoot currentPath numberedArgs patterns segments = runExceptT do
 
 -- Expand a numeric argument like `1` or a range like `3-9`
 expandNumber :: [String] -> String -> [String]
-expandNumber numberedArgs s =
-  maybe
-    [s]
-    (map (\i -> fromMaybe (show i) . atMay numberedArgs $ i - 1))
-    expandedNumber
+expandNumber numberedArgs s = case expandedNumber of
+  Nothing -> [s]
+  Just nums ->
+    [s | i <- nums, Just s <- [vargs Vector.!? (i - 1)]]
   where
+    vargs = Vector.fromList numberedArgs
```

Comment:
A couple things:

1. The old code had a quadratic performance bug, where it was asking for the 1st index of a list, then the second, then the third, and so on. But list indexing takes linear time! So I've converted the args to a vector.
2. This new code allows you to ask for more args than are available, and it'll just ignore any numbers that don't exist. This is handy for saying `edit 1-1000` and not having to worry about whether the list has exactly 1000 elements. I use this for the round trip transcripts so I don't have to keep changing the number passed to the `edit` command.

Aside, the syntax `edit 1-` could be cool, it fills in the ending number for you. But I didn't trust myself to edit the regex code here. :)